## Домашнее задание №4
***

### Задача 1
Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

In [1]:
import numpy as np

In [46]:
from statistics import NormalDist

#### Решение
По условию задачи, случайная непрерывная величина А имеет равномерное распределение на отрезке $(a;b]$\
Для рассчета средней арифметической (математического ожидания) используем формулу:\
$\displaystyle\text {Математическое ожидание: }M(X)=\frac{a+b}{2}$  

In [47]:
def Get_average_value(a, b):
    return (a + b)/2

In [48]:
print(f'Среднее значение М(А) = {Get_average_value(200, 800): .2f}')

Среднее значение М(А) =  500.00


для расчета дисперсии, воспользуемся формулой: $\\\displaystyle\text {Дисперсия: }D(X)=\frac{(b-a)^2}{12}$ 

In [49]:
def Get_variance(a, b):
    return ((b - a)**2)/12

In [50]:
print(f'Дисперсия D(А) = {Get_variance(200, 800): .2f}')

Дисперсия D(А) =  30000.00


---
### Ответ:
**Среднее арифметическое М(А) = 500.00**\
**Дисперсия D(A) = 30 000.00**

---

### Задача 2
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.

#### Решение
По условию задачи имеем: равномерное распределение, $D(B)=0.2$ и $a=0.5$  
Подставив значения в формулу дисперсии найдем правую границу $b$ 
$\displaystyle \frac{(b-0.5)^2}{12}=0.2\\(b-0.5)^2=0.2*12\\b-0.5=\sqrt{2.4}\\b=0.5+\sqrt{2.4}$  

In [51]:
b = 0.5 + 2.4**(1/2)
print(f'Правая граница непрерывной величины В = {b: .3f}')

Правая граница непрерывной величины В =  2.049


Зная правую границу, найдем среднее арифеметическое:  
$\displaystyle M(B)=\frac{a+b}{2}$

In [52]:
print(f'Среднее значение М(А) = {Get_average_value(0.5, b): .3f}')

Среднее значение М(А) =  1.275


---
### Ответ:
**Правая граница непрерывной величины В = 2.049**\
**Среднее арифметическое М(В) = 1.275**

---

### Задача 3
Непрерывная случайная величина X распределена нормально и задана плотностью распределения\
 $\displaystyle f(x)=\frac{1}{4\cdot\sqrt{2\pi}}\cdot e^{-\frac{(x+2)^2}{32}}$,\
 Найдите:  
   а) $M(X)$  
   б) $D(X)$  
   в) $\displaystyle\sigma$ (среднее квадратичное отклонение)

#### Решение
В функции плотности распределения уже подставлены параметры, выберем их:

---  
#### Ответ:  
$M(X)=-2\\D(X)=16\\\displaystyle\sigma=4$  

---

### Задача 4
Рост взрослого населения города X имеет нормальное распределение.
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.

Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:\
а) больше 182 см\
б) больше 190 см\
в) от 166 см до 190 см\
г) от 166 см до 182 см\
д) от 158 см до 190 см\
е) не выше 150 см или не ниже 190 см\
ё) не выше 150 см или не ниже 198 см\
ж) ниже 166 см.

#### Решение
Для решения задачи, воспользуемся формулой $\displaystyle\text {Z - значения: }Z=\frac{X-\mu}{\sigma}$\
соответствующие значения вероятности можно посмотреть на этом [ресурсе](https://datascience.eu/ru/%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0-%D0%B8-%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D0%BA%D0%B0/z-%D1%82%D0%B0%D0%B1%D0%BB%D0%B8%D1%86%D0%B0/)\
 Для расчет вероятности воспользуемся методом cdf, объекта NormalDist библиотеки statistics


In [72]:
def Get_z_probability(x, m, s):
    return NormalDist().cdf((x - m)/s)

In [94]:
print(f'Вероятность, что рост случайным образом выбранного человека больше 182 см = {1 - Get_z_probability(182, 174, 8): .4f}')

Вероятность, что рост случайным образом выбранного человека больше 182 см =  0.1587


In [84]:
print(f'Вероятность, что рост случайным образом выбранного человека больше 190 см = {1 - Get_z_probability(190, 174, 8): .4f}')

Вероятность, что рост случайным образом выбранного человека больше 190 см =  0.0228


In [85]:
print(f'Вероятность, что рост случайным образом выбранного человека от 166 см до 190 см = {Get_z_probability(190, 174, 8) - Get_z_probability(166, 174, 8): .4f}')

Вероятность, что рост случайным образом выбранного человека от 166 см до 190 см =  0.8186


In [86]:
print(f'Вероятность, что рост случайным образом выбранного человека от 166 см до 182 см = {Get_z_probability(182, 174, 8) - Get_z_probability(166, 174, 8): .4f}')

Вероятность, что рост случайным образом выбранного человека от 166 см до 182 см =  0.6827


In [87]:
print(f'Вероятность, что рост случайным образом выбранного человека от 158 см до 190 см = {Get_z_probability(190, 174, 8) - Get_z_probability(158, 174, 8): .4f}')

Вероятность, что рост случайным образом выбранного человека от 158 см до 190 см =  0.9545


In [91]:
print(f'Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 190 см = {Get_z_probability(150, 174, 8) + (1 - Get_z_probability(190, 174, 8)) : .4f}')

Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 190 см =  0.0241


In [92]:
print(f'Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 198 см = {Get_z_probability(150, 174, 8) + (1 - Get_z_probability(198, 174, 8)) : .4f}')

Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 198 см =  0.0027


In [93]:
print(f'Вероятность, что рост случайным образом выбранного человека ниже 166 см = {Get_z_probability(166, 174, 8) : .4f}')

Вероятность, что рост случайным образом выбранного человека ниже 166 см =  0.1587


---  
#### Ответ:  
**а) Вероятность, что рост случайным образом выбранного человека больше 182 см =  0.1587\
б) Вероятность, что рост случайным образом выбранного человека больше 190 см =  0.0228\
в) Вероятность, что рост случайным образом выбранного человека от 166 см до 190 см =  0.8186\
г) Вероятность, что рост случайным образом выбранного человека от 166 см до 182 см =  0.6827\
д) Вероятность, что рост случайным образом выбранного человека от 158 см до 190 см =  0.9545\
е) Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 190 см =  0.0241\
ж) Вероятность, что рост случайным образом выбранного человека не выше 150 см или не ниже 198 см =  0.0027\
з) Вероятность, что рост случайным образом выбранного человека ниже 166 см =  0.1587**

---

### Задача 5
 На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

#### Решение
Для решения задачи, воспользуемся формулой $\displaystyle\text {Z - значения: }Z=\frac{X-\mu}{\sigma}$

In [98]:
quantity = (190-178)/25**(1/2)
print(f'Рост человека 190 см отклоняется от математического ожидания в популяции в которой M(X) = 178 см и D(X) = 25 кв.см на {quantity} сигмы')

Рост человека 190 см отклоняется от математического ожидания в популяции в которой M(X) = 178 см и D(X) = 25 кв.см на 2.4 сигмы


---  
#### Ответ:  
Рост человека 190 см отклоняется от математического ожидания в популяции в которой M(X) = 178 см и D(X) = 25 кв.см на 2.4 сигмы

---